### ISSUE TITLE
NCC loss increases sometimes

### ISSUE THREAD

**Issue by xiaowk5516** [voxelmorph/voxelmorph#158]
Hi! I run train.py with IXI dataset.I use ncc as loss function and set lambda 1.but i find the loss like this:

> Epoch 1/100
100/100 [==============================] - 171s 2s/step - loss: -0.1610 - spatial_transformer_1_loss: -0.2132 - flow_loss: 0.0521
Epoch 2/100
100/100 [==============================] - 144s 1s/step - loss: -0.1699 - spatial_transformer_1_loss: -0.2224 - flow_loss: 0.0525
Epoch 3/100
100/100 [==============================] - 144s 1s/step - loss: -0.1741 - spatial_transformer_1_loss: -0.2261 - flow_loss: 0.0520
Epoch 4/100
100/100 [==============================] - 143s 1s/step - loss: -0.1770 - spatial_transformer_1_loss: -0.2286 - flow_loss: 0.0515
Epoch 5/100
100/100 [==============================] - 143s 1s/step - loss: -0.1712 - spatial_transformer_1_loss: -0.2241 - flow_loss: 0.0528
Epoch 6/100
100/100 [==============================] - 144s 1s/step - loss: -0.1790 - spatial_transformer_1_loss: -0.2304 - flow_loss: 0.0514
Epoch 7/100
100/100 [==============================] - 161s 2s/step - loss: -0.1795 - spatial_transformer_1_loss: -0.2303 - flow_loss: 0.0508
Epoch 8/100
100/100 [==============================] - 144s 1s/step - loss: -0.1767 - spatial_transformer_1_loss: -0.2287 - flow_loss: 0.0520
Epoch 9/100
100/100 [==============================] - 144s 1s/step - loss: -0.1844 - spatial_transformer_1_loss: -0.2340 - flow_loss: 0.0497
Epoch 10/100
100/100 [==============================] - 144s 1s/step - loss: -0.1899 - spatial_transformer_1_loss: -0.2383 - flow_loss: 0.0484
Epoch 11/100
100/100 [==============================] - 144s 1s/step - loss: -0.1807 - spatial_transformer_1_loss: -0.2317 - flow_loss: 0.0510
Epoch 12/100
100/100 [==============================] - 144s 1s/step - loss: -0.1820 - spatial_transformer_1_loss: -0.2326 - flow_loss: 0.0506

why dose the loss increase sometimes?

when i use mse as loss funtion, this would not happen.

> Epoch 1/100
100/100 [==============================] - 118s 1s/step - loss: 0.0025 - spatial_transformer_1_loss: 0.0018 - flow_loss: 0.0680
Epoch 2/100
100/100 [==============================] - 110s 1s/step - loss: 0.0024 - spatial_transformer_1_loss: 0.0017 - flow_loss: 0.0709
Epoch 3/100
100/100 [==============================] - 108s 1s/step - loss: 0.0024 - spatial_transformer_1_loss: 0.0017 - flow_loss: 0.0705
Epoch 4/100
100/100 [==============================] - 111s 1s/step - loss: 0.0024 - spatial_transformer_1_loss: 0.0017 - flow_loss: 0.0729
Epoch 5/100
100/100 [==============================] - 105s 1s/step - loss: 0.0023 - spatial_transformer_1_loss: 0.0016 - flow_loss: 0.0690

**Comment by studentguoyiyuan** [comment]
Hi, have you solved it ? Does it matter?

**Comment by adalca** [comment]
@xiaowk5516 vxm uses *stochastic* gradient methods to optimize, so sometimes the loss will go up. -- what matters is if it goes down over time as a trend.

**Comment by xiaowk5516** [comment]
> @xiaowk5516 vxm uses _stochastic_ gradient methods to optimize, so sometimes the loss will go up. -- what matters is if it goes down over time as a trend.

Thanks!
